### PR TITLE
[ci] [backend] Backend tests in travis

### PR DESCRIPTION
**Pull request by eduardoj**
Option "TEST_SUITE=backend" added in .travis.yml file, so perl backend tests can be run with travis.

TEST_SUITE variable is passed as argument to dist/ci/obs_testsuite_travis_install.sh, so ubuntu package libdevel-cover-perl is installed only when testing backend.

### CODE REVIEW

**Comment by bgeuken** [comment]
Could you remove those two commits: 41754288856f33642c and 9830de1c3b9a07add? Since those are just temporary changes.

Rest looks good to me:+1:

@adrianschroeter @mlschroe @M0ses What do you say?

**Comment by mdeniz** [comment]
Agree with @bgeuken in killing temporary changes.
LGTM :+1:

**Comment by adrianschroeter** [comment]
yep, we can merge after those are gone.

**Comment by mdeniz** [comment]
@M0ses or @mlschroe the only nitpick I see is that Travis is failing in one test on the backend test suite... this one https://travis-ci.org/openSUSE/open-build-service/jobs/120339283#L1292

File: dist/ci/obs_testsuite_travis_install.sh [hunk 1]
```diff
@@ -17,3 +19,7 @@ sudo apt-get update
 
 # Install the dependencies of the backend
 sudo apt-get install --force-yes travis-deps libxml-parser-perl libfile-sync-perl python-rpm python-urlgrabber python-sqlitecachec python-libxml2 createrepo libbssolv-perl sphinxsearch libjson-xs-perl libxml-simple-perl libgd-gd2-perl
+
+if [ "$TEST_SUITE" == "backend" ]; then
+  sudo apt-get install --force-yes libdevel-cover-perl
+fi
```
Comment (hennevogel) [inline dist/ci/obs_testsuite_travis_install.sh]: If you're already conditionally install packages per test suite, why don't you skip the parts not needed for the backend test suite? :-)

**Comment by eduardoj** [comment]
@bgeuken: commits 4175428 and 9830de1 removed. Branch changed by "git rebase ..." and "git push -f".

Thank you all for the comments.

**Comment by M0ses** [comment]
@eduardoj : First of all - Thanks for your work on this.

Before we could merge this PR, the Problem with the broken test must be fixed. IMO it would not make much sense to activate a broken test suite.

The problem here seems not to be any of the test case - they finish with "ok" - but the cleanup mechanism dies because a file ( t/data/0015/events/x86_64/package:project-name::test-package-name ) is not there. The question for me is, why is this file missing while the travis checks and not while running the tests on a freshly installed vm or e.g. while package  building in obs.

**Comment by eduardoj** [comment]
@MOses: Hi!

Maybe in line 382 of the spec file ( https://build.opensuse.org/package/view_file/OBS:Server:Unstable/obs-server/_service:extract_file:obs-server.spec ) this line:

`make -C src/backend test`

must be:

`make -C src/backend test_build`

Please, have a look at https://github.com/openSUSE/open-build-service/blob/master/src/backend/Makefile#L31

**Comment by M0ses** [comment]
@eduardoj: Hi :-) 

the "test" target is (IMHO) completly ok there 

the "test" target in the Makefile is some kind of alias to "test_unit" included from
https://github.com/openSUSE/open-build-service/blob/master/Makefile.include#L16

which will run a test_unit  in 
https://github.com/openSUSE/open-build-service/blob/master/src/backend/Makefile#L31

The result is the same:

```
cd src/backend
make test_unit
```

**Comment by adrianschroeter** [comment]
Just had a discussion this morning with Frank. This pull request itself is fine, but the problem is the failure which happens only on travis and no where else.

We can't spend time atm to find out why the backend test suite only breaks on travis and nowhere else. So we just leave this pull request open for now.... Maybe someone else likes to debug that :)

**Comment by hennevogel** [comment]
[[] Merge PullRequest #1672](https://trello.com/c/IxyUmrwM/16-merge-pullrequest-1672)

**Comment by eduardoj** [comment]
Backend test in travis makes no much sense until backend is ready to work in Ubuntu.

**Pull request closed without merging**